### ISSUE TITLE
Show Upcoming Events and Tasks

### ISSUE THREAD

**Issue by danrabbit** [elementary/wingpanel-indicator-datetime#293]
### Problem

Sometimes I have a task with a due by date of tomorrow or a couple days from now. Or I have an event taking place tomorrow or in a couple days that I need to be prepared for.

Currently we only show tasks and events from today in the events pane, so unless you click over to the next day in the calendar, these upcoming events and tasks are out of sight and mind

### Proposal

I think it would be helpful to show upcoming events and tasks for upcoming events in addition to events that are happening today

### Prior Art (Optional)

In calendar, we show upcoming events broken into sections like "this week" etc. We probably don't need to go as far as we do in Calendar, but showing upcoming events and tasks for the week would be helpful

**Comment by cscottyb** [comment]
### Feature requests
- Always visible scroll bar to see how long of a list you have for the week.
- Toggle a task's 'checkbox' in the panel, but it will remain in the Today view until the next day.
- Events and tasks could look distinct from each other for glanceable information. 


### Grouping and Sorting Hierarchy of Events and Tasks per-day: 
- Multiday Events (vacation)
- All-day Events (birthdays, holidays)
- Scheduled Events (appointments, classes)
- Scheduled incomplete Tasks
- Today's Completed Tasks
- ~~Unscheduled Tasks~~


![nowpanel](https://user-images.githubusercontent.com/7649906/144685234-34c12204-525a-4be5-85f9-23f9d3a8fac0.png)
This mock-up, for a future proposal, includes some feature requests ideas listed above. Note that the Events sorting is backwards, but I think Calendar events should go grouped before tasks. This mockup is far, far from finish.

**Comment by marbetschar** [comment]
> ### Problem
> 
> Sometimes I have a task with a due by date of tomorrow or a couple days from now. Or I have an event taking place tomorrow or in a couple days that I need to be prepared for.


> ### Proposal

**TL;DR**

For this use case to work, we can simply allow the user to define different start and due in the Tasks app for a task. In Wingpanel, we then make sure the task is shown on each day covered by this timespan.

**Technical background**

In terms of the [CalDAV standard](https://icalendar.org/iCalendar-RFC-5545/3-6-2-to-do-component.html), there is the possibility to define a different `DTSTART` property value than for `DUE`. Currently we set both fields to the same value, [because EDS did not support querying for `DUE`](https://github.com/elementary/tasks/pull/147). This has been fixed, so on OS7 we are no longer "forced" to set both fields to the same value.

> ### Prior Art (Optional)
> 
> In calendar, we show upcoming events broken into sections like "this week" etc. We probably don't need to go as far as we do in Calendar, but showing upcoming events and tasks for the week would be helpful

Personally I don't like the way this is done in Calendar. For me it is very confusing and I'm still not able to wrap my head around this, because I primarily select a day in the grid and intuitively expect everything in the right sidebar to be relative to the selected day.

Also, when I look at a date which is far in the future or past having a glimpse at "Tomorrow" does not really has any benefit for me.
